query/src/main/java/org/infinispan/query/dsl/embedded/impl/BaseJPAFilterIndexingServiceProvider.java

```diff
@@ -200,8 +207,10 @@ void unregister() {
       @Override
       public void onFilterResult(Object userContext, Object eventType, Object instance, Object[] projection, Comparable[] sortProjection) {
          CacheEntryEvent<K, V> event = (CacheEntryEvent<K, V>) userContext;
+         Object storedKey = wrapper.wrapKey(encoder.keyToStorage(event.getKey()));
+         boolean primaryKey = clusteringDependentLogic.getCacheTopology().getDistribution(storedKey).isPrimary();
```

Comment:
Let me revisit this again change again, can't remember exactly what was the issue